tools/Update-Progress.ps1

```diff
@@ -0,0 +1,36 @@
+# Update-Progress.ps1
+
+function Update-Progress {
+    param (
+        [Parameter(Mandatory, Position = 0)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StatusMessage,
+
+        [Parameter(Mandatory, Position = 1)]
+        [ValidateRange(0, 100)]
+        [int]$Percent,
+
+        [Parameter(Position = 2)]
+        [string]$Activity,
+
+        [Parameter(Position = 3)]
+        [switch]$LogProgress
+    )
+
+    # Default activity to "Processing" if not provided
+    if (-not $Activity) {
+        $Activity = "Processing"
+    }
+
+    # Write the progress to the console
+    Write-Progress -Activity $Activity -Status $StatusMessage -PercentComplete $Percent
+
+    # Optionally log the progress to a file
+    if ($LogProgress) {
+        $logMessage = "{0:yyyy-MM-dd HH:mm:ss} - {1} - {2}% - {3}" -f (Get-Date), $Activity, $Percent, $StatusMessage
+        $logMessage | Out-File -FilePath "$PSScriptRoot\progress.log" -Append -Encoding utf8
+    }
+}
+
+# Example Usage:
+# Update-Progress "Processing files..." 50 "File Processing" -LogProgress
```

Comment:
I recommend changing this documentation to be set in the beginning of `Update-Progress` function, it's considered PowerShell standards to do so, and follows the general coding style of this project. A quick example on this is the following:
```ps1
function Update-Progress {
    <#
        .SYNOPSIS
            A wrapper for PowerShell 'Write-Progress' Cmdlet.
        .PARAMETER StatusMessage
            A mandatory parameter which'll be used when displaying progress bar using 'Write-Progress' Cmdlet.
        .PARAMETER Percent
            # Fill in rest of documentation for parameters here...
        .EXAMPLE
            Update-Progress "Processing files..." 50 "File Processing" -LogProgress
    #>

    # Parameters here
    # ...

    # Rest of the code here
    # ...
}
```

Reply:
I have edited according to your criteria that I did not do before. Thanks for pointing it out.